temporal_sqlalchemy/clock.py

```diff
@@ -221,6 +195,46 @@ def build_clock_class(
     return type('%sClock' % name, base_classes, props)
 
 
+def build_clock_table(entity_table: sa.Table,
+                      metadata: sa.MetaData,
+                      schema: str,
+                      activity_class=None) -> sa.Table:
+    clock_table_name = util.truncate_identifier(
+        "%s_clock" % entity_table.name)
+    clock_table = sa.Table(
+        clock_table_name,
+        metadata,
+        sa.Column('tick',
+                  sa.Integer,
+                  primary_key=True,
+                  autoincrement=False),
+        sa.Column('timestamp',
+                  sa.DateTime(True),
+                  server_default=sa.func.current_timestamp()),
+        schema=schema)
+
+    entity_keys = set()
+    for fk in util.foreign_key_to(entity_table, primary_key=True):
+        # this is done to support arbitrary primary key shape on entity
+        clock_table.append_column(fk)
+        entity_keys.add(fk.key)
+
+    if activity_class:
+        activity_keys = set()
+        # support arbitrary shaped activity primary keys
+        for fk in util.foreign_key_to(activity_class.__table__,
+                                      prefix='activity',
+                                      nullable=False):
+            clock_table.append_column(fk)
+            activity_keys.add(fk.key)
+        # ensure we have DB constraint on clock <> activity uniqueness
+        clock_table.append_constraint(
+            sa.UniqueConstraint(*(entity_keys | activity_keys))
+        )
```

Comment:
@bijanvakili and @cvlong this is where we can fix the clock <> activity uniqueness, though it'll generate a lot of migrations for us, I think it's the right call.